vgscollect/src/main/java/com/verygoodsecurity/vgscollect/VGSCollectLogger.kt

```diff
@@ -0,0 +1,71 @@
+package com.verygoodsecurity.vgscollect
+
+import android.util.Log
+
+/**
+ * This object is used to log messages in VGS Collect SDK.
+ */
+object VGSCollectLogger {
+
+    /** Current priority level for filtering debugging logs */
+    var logLevel: Level = if(BuildConfig.DEBUG) Level.DEBUG else Level.NONE
+        @JvmName("setLogLevel") set
+        @JvmName("getLogLevel") get
+
+
+    /**
+     * Priority constant for the printing debug-logs.
+     */
+    enum class Level {
+
+        /**
+         * Default setting. We print all information about processing.
+         * It includes errors, warnings, notifications, debug messages, requests and responses.
+         */
+        DEBUG,
+
+        /**
+         * This setting allows you to minimize information and print only errors, warnings.
+         */
+        WARN,
+
+        /**
+         * Disable debug-logs.
+         */
+        NONE
+    }
+
+    /**
+     * Returns true if the logger print log messages.
+     */
+    fun isDebugEnabled(): Boolean = logLevel.ordinal != Level.NONE.ordinal
+
+    private fun printLog(level: Level, tag: String, message: String) {
+        if(level.ordinal >= logLevel.ordinal) {
+            when (level) {
+                Level.DEBUG -> Log.d(tag, message)
+                Level.WARN -> Log.w(tag, message)
+            }
+        }
+    }
+
+    /**
+     * Print a WARN log message.
+     *
+     * @param tag Used to identify the source of a log message.
+     * @param message The message you would like logged.
+     */
+    fun warn(tag: String, message: String) {
```

Comment:
@Endorf I think it's better to make warn/debug private for two reasons:

1. If dev sets the tag too long it will throw an exception.
2. It will be much easier to maintain this class if dev will not be able to use it and I think he should not use it.